src/main/java/net/imagej/ops/threshold/rosin/ComputeRosinThreshold.java

```diff
@@ -0,0 +1,187 @@
+                int pk, st, fi, d1, d2;
+                long tmp;
+                int i;
+                int thresh;
+                boolean doInvert = false;
+
+                pk = find_start(hist, no_pts);
+                st = find_first(hist, no_pts);
+                fi = find_end(hist, no_pts);
+                d1 = pk - st;
+                d2 = fi - pk;
+
+                if ((d1 < 0) || (d2 < 0)) {
+                        log.warn("Histogram peak at a strange location");
+                }
+
+                if (d1 > d2) {
+                        doInvert = true;
+                }
+
+                if (doInvert) {
+                        for (i = 0; i < no_pts / 2; i++) {
+                                tmp = hist[i];
+                                hist[i] = hist[no_pts - 1 - i];
+                                hist[no_pts - 1 - i] = tmp;
+                        }
+                }
+
+                st = find_start(hist, no_pts);
+
+                thresh = find_corner(hist, st, no_pts);
+
+                if (doInvert) {
+                        thresh = no_pts - thresh - 1;
+                }
+
+                return thresh;
+        }
+
+        private int find_start(long[] Y, int no_pts) {
+                int i, st;
+                long max;
+
+                // Find largest peak
+                st = 0;
+                max = Y[0];
+                for (i = 1; i < no_pts; i++) {
+                        if (Y[i] > max) {
+                                max = Y[i];
+                                st = i;
+                        }
+                }
+
+                return st;
+        }
+
+        private int find_first(long[] Y, int no_pts) {
+                int i, st;
+
+                st = 0;
+                for (i = 0; i < no_pts; i++) {
+                        if (Y[i] > 0) {
+                                st = i;
+                                break;
+                        }
+                }
+
+                return st;
+        }
+
+        private int find_end(long[] Y, int no_pts) {
+                int i, fi;
+
+                fi = 0;
+                for (i = 1; i < no_pts; i++) {
+                        if (Y[i] > 0) {
+                                fi = i;
+                        }
+                }
+
+                return fi;
+        }
+
+        private int find_corner(long[] Y, int st, int no_pts) {
+
+                long[] X = new long[no_pts];
+                int i;
+                float dist;
+                float max_dist = -1;
+                int thresh = -1;
+                int end;
+
+                for (i = st; i < no_pts; i++) {
+                        X[i] = i;
+                }
+
+                end = no_pts - 1;
+                while ((Y[end] == 0) && (end >= 0)) {
+                        end--;
+                }
+                no_pts = end;
+                if (end <= 0) {
+                        log.error("Histogram is empty.");
```

Comment:
How do we handle this situation? when can it occur? for now you can maybe throw a `IllegalStateException`. Then you can also get rid of the logger for now.